cabal-install/src/Distribution/Client/CmdTarget.hs

```diff
@@ -0,0 +1,208 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Distribution.Client.CmdTarget
+  ( targetCommand
+  , targetAction
+  ) where
+
+import Distribution.Client.Compat.Prelude
+import Prelude ()
+
+import qualified Data.Map as Map
+import Distribution.Client.CmdBuild (selectComponentTarget, selectPackageTargets)
+import Distribution.Client.CmdErrorMessages
+import Distribution.Client.InstallPlan
+import qualified Distribution.Client.InstallPlan as InstallPlan
+import Distribution.Client.NixStyleOptions
+  ( NixStyleFlags (..)
+  , defaultNixStyleFlags
+  , nixStyleOptions
+  )
+import Distribution.Client.ProjectOrchestration
+import Distribution.Client.ProjectPlanning
+import Distribution.Client.Setup
+  ( ConfigFlags (..)
+  , GlobalFlags
+  )
+import Distribution.Client.TargetProblem
+  ( TargetProblem'
+  )
+import Distribution.Package
+import Distribution.Simple.Command
+  ( CommandUI (..)
+  , usageAlternatives
+  )
+import Distribution.Simple.Flag (fromFlagOrDefault)
+import Distribution.Simple.Utils
+  ( noticeDoc
+  , safeHead
+  , wrapText
+  )
+import Distribution.Verbosity
+  ( normal
+  )
+import Text.PrettyPrint
+import qualified Text.PrettyPrint as Pretty
+
+-------------------------------------------------------------------------------
+-- Command
+-------------------------------------------------------------------------------
+
+targetCommand :: CommandUI (NixStyleFlags ())
+targetCommand =
+  CommandUI
+    { commandName = "v2-target"
+    , commandSynopsis = "Disclose selected targets."
```

Comment:
"Disclose" means "to reveal information that was previously hidden / kept secret". I don't think this is the correct word. `Display` or `print` would be better fits.

Reply:
Both display and print are very generic.

> disclose:
> To uncover and expose to view (anything material); to remove a covering from; to reveal, allow to be seen.
> SOURCE: [OED, disclose](https://www.oed.com/dictionary/disclose_v?tl=true)

Reply:
I've changed the synopsis to use target as a verb:

```diff
-    , commandSynopsis = "Disclose selected targets."
+    , commandSynopsis = "Target a subset of all targets."
```

Reply:
@philderbeast Yes they are very generic, because we are doing a generic operation. We are not exposing to view something previously hidden, as is the meaning of disclose. :)

Actually I much prefer how you phrased it in the description of the PR:

> showing targets that can be supplied to other commands

If you could change the other instances of `disclose` that would be grand. This is my only remark regarding this PR. :)